fix: memory leak: def_urlpre is not freed

Resolves: #816168
Upstream commit: e0adc1aa

diff a/clients/tools/ldapsearch.c b/clients/tools/ldapsearch.c
--- a/clients/tools/ldapsearch.c
+++ b/clients/tools/ldapsearch.c
@@ -1334,6 +1334,11 @@ getNextPage:
 	if ( derefval.bv_val != NULL ) {
 		ldap_memfree( derefval.bv_val );
 	}
+	if ( urlpre != NULL ) {
+		if ( def_urlpre != urlpre )
+			free( def_urlpre );
+		free( urlpre );
+ 	}
 
 	if ( c ) {
 		for ( ; save_nctrls-- > 0; ) {
--
